DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Processors/SubstituteProcessor.cs

```diff
@@ -0,0 +1,73 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Anonymizer.Core.Exceptions;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Processors
+{
+    public class SubstituteProcessor : IAnonymizerProcessor
+    {
+        private readonly string _replaceWith = "Anonymous";
+
+        public SubstituteProcessor(JObject settingObject)
+        {
+            EnsureArg.IsNotNull(settingObject, nameof(settingObject));
+
+            if (settingObject.TryGetValue("replaceWith", StringComparison.OrdinalIgnoreCase, out JToken replaced))
+            {
+                _replaceWith = replaced.ToString();
+            }
+        }
+
+        public void Process(DicomDataset dicomDataset, DicomItem item, ProcessContext context = null)
+        {
+            EnsureArg.IsNotNull(dicomDataset, nameof(dicomDataset));
+            EnsureArg.IsNotNull(item, nameof(item));
+
+            try
+            {
+                if (item.ValueRepresentation == DicomVR.OW && !(item is DicomFragmentSequence))
+                {
+                    dicomDataset.AddOrUpdate(item.ValueRepresentation, item.Tag, ushort.Parse(_replaceWith));
+                }
+                else if (item.ValueRepresentation == DicomVR.OL)
+                {
+                    dicomDataset.AddOrUpdate(item.ValueRepresentation, item.Tag, uint.Parse(_replaceWith));
```

Comment:
>uint.Parse(_replaceWith) [](http://example.com/codeflow?start=81&length=24)

since you are using this string as a number, why set its default value as "Anonymous"? A "0" is a better value...

Reply:
The value for DICOM elements are string most times.

Reply:
try to use two default values, one for string and another for number types.